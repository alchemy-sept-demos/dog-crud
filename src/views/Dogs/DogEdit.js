import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import DogForm from '../../components/dogs/DogForm';
import { fetchDogById, updateDog } from '../../services/dogs';

export default function DogEdit() {
  // const [dog, setDog] = useState({});
  const [name, setName] = useState('');
  const [bio, setBio] = useState('');
  const [image, setImage] = useState('');
  const [age, setAge] = useState('');
  const [breed, setBreed] = useState('');

  const params = useParams();
  useEffect(() => {
    const fetchData = async () => {
      const resp = await fetchDogById(params.id);

      setName(resp.name);
      setBio(resp.bio);
      setImage(resp.image);
      setAge(resp.age);
      setBreed(resp.breed);
    };
    fetchData();
  }, [params.id]);

  // const updateDog = (key, value) => {
  //   dog[key] = value; // updating an individual key
  //   setDog({ ...dog }); // setting the change into state
  // };
  const handleSubmit = async (e) => {
    e.preventDefault();
    const data = await updateDog(params.id, name, bio, image, breed);
  };

  return (
    <div>
      <h1>Edit Dog</h1>
      <DogForm
        name={name}
        setName={setName}
        bio={bio}
        setBio={setBio}
        image={image}
        setImage={setImage}
        age={age}
        setAge={setAge}
        breed={breed}
        setBreed={setBreed}
        handleSubmit={handleSubmit}
      />
    </div>
  );
}
