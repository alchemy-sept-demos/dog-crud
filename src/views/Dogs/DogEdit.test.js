import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import DogEdit from './DogEdit';

it("should render the dog's information in a form", async () => {
  // render
  const { container } = render(
    // MemoryRouter is needed because <DogEdit> uses the <Link> component
    <MemoryRouter initialEntries={['/dogs/4/edit']}>
      <Route path="/dogs/:id/edit" component={DogEdit} />
    </MemoryRouter>
  );
  // wait for stuff to load
  await screen.findByDisplayValue('Franny');
  // take a snapshot
  expect(container).toMatchSnapshot();
});
