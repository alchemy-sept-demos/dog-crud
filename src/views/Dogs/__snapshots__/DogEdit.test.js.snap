// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the dog's information in a form 1`] = `
<div>
  <div>
    <h1>
      Edit Dog
    </h1>
    <div
      class="dog-form"
    >
      <form>
        <h2>
          Enter Dog Details
        </h2>
        <div
          class="form-control"
        >
          <label>
            Name:
          </label>
          <input
            placeholder="Name"
            type="text"
            value="Franny"
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Bio:
          </label>
          <textarea
            placeholder="Bio"
          >
            Sweet Frenchie with lots of love to give.
          </textarea>
        </div>
        <div
          class="form-control"
        >
          <label>
            Age: 
          </label>
          <input
            placeholder="Age"
            type="number"
            value="4"
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Image: 
          </label>
          <input
            placeholder="Image URL"
            type="text"
            value="https://placedog.net/500?id=51"
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Breed: 
          </label>
          <input
            placeholder="Breed"
            type="text"
            value="French Bulldog"
          />
        </div>
        <div
          class="action"
        >
          <button>
            Save
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
